### PR TITLE
feat: adds integration testing module

### PR DESCRIPTION
**Pull request by malachi-constant**
### Module: `integration-tests`
- Adds CodePipeline that runs `seedfarmer apply` & `seedfarmer destroy` against a list of provided manifests from your repository.

### Consideration
**Please see module's readme for overview before reviewing the rest of the module**
1. This is a *starting point*. We can continue to iterate on this module to support different use cases. E.g. (scheduled tests, alternative vcs sources, custom validation)
2. This is a module to be deployed alongside a solution. The integration tests will run tests of all module groups beside itself. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: b290489583d832991047181ad82640b8610d48c9
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=4a8a4428-3089-4d7c-8cd0-305fa2fcf1de%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: f5afb81535aeb948f0d353a192bac65cc7734d21
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=bc66a8be-f3f4-46b9-84b6-181459215347%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: 2d21d800dcd01734b94079671c2cf47df61d0d3e
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=e9ee31ac-6815-4eed-bf69-217947edfc2e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: modules/testing/integration-tests/README.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+[Example deployment manifest using an alias for integration tests region](../../../manifests/example-integration-tests/deployment.yaml):
+
+```yaml
+name: example-integration-tests
+toolchainRegion: us-west-2
+forceDependencyRedeploy: True
+groups:
+  - name: optionals
+    path: manifests/example-integration-tests/optional-modules.yaml
```
Comment (srinivasreddych) [inline modules/testing/integration-tests/README.md:41]: `optional-modules.yaml` file is missing in the manifests dir. Did you mean `core-modules.yaml` instead?

File: modules/testing/integration-tests/README.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+seedfarmer bootstrap toolchain --project idf --trusted-principal ${ROLE_ARN} --region ${AWS_REGION} --as-target
+```
+
+2. Deploy Manifest
+```bash
+seedfarmer apply manifests/example-integration-tests/deployment.yaml
+```
+
+3. Once the stack is deployed... ![Integ Stack](./images/integ_stack_deployed.png)
+push code to the repo and branch specified in your module to start the integration test.
```
Comment (srinivasreddych) [inline modules/testing/integration-tests/README.md:89]: Can you elaborate on what code needs to be pushed and to which repo?

**Review by srinivasreddych — DISMISSED** [review]

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: a217c4d3fe4295ec5a344f9932acc171f2716c55
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=cc069490-c138-44f7-a3e2-e06496bf64d2%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by dgraeber — APPROVED** [review]
Looks good...do we need a changelog

**Pull request merged**